### PR TITLE
[MoonReports]: Excel adds some tabs to the group line

### PR DESCRIPTION
**Pull request by peerau**
this adds them to the regex, without removing compat to EVE groups

### CODE REVIEW

**Comment by warlof** [comment]
Hi, thanks for this update.
Can you also provide a test which is covering that change ?

**Comment by peerau** [comment]
Tests have been updated to use dataSources, and checked against the new input types.

File: tests/ReportParser/Parsers/MoonReportTest.php [hunk 1]
```diff
@@ -223,9 +280,12 @@ public function testGetGroups()
         $this->assertContainsOnlyInstancesOf(Group::class, $report->getGroups());
     }
 
-    public function testHasElements()
+    /**
+     * @dataProvider correctData
+     */
+    public function testHasElements(string $path)
```
Comment (warlof) [inline tests/ReportParser/Parsers/MoonReportTest.php:286]:
```suggestion
    public function testHasElements(string $artifact_path)
```

File: tests/ReportParser/Parsers/MoonReportTest.php [hunk 2]
```diff
@@ -223,9 +280,12 @@ public function testGetGroups()
...
     {
-        $content = file_get_contents(__DIR__ . '/../../artifacts/moon_report.txt');
+        $content = file_get_contents(__DIR__ . $path);
```
Comment (warlof) [inline tests/ReportParser/Parsers/MoonReportTest.php:288]:
```suggestion
        $content = file_get_contents(__DIR__ . $artifact_path);
```

File: tests/ReportParser/Parsers/MoonReportTest.php [hunk 3]
```diff
@@ -202,19 +253,25 @@ public function testGetHeader()
...
     }
 
-    public function testGetGroups()
+    /**
+     * @dataProvider correctData
+     */
+    public function testGetGroups(string $path)
     {
-        $content = file_get_contents(__DIR__ . '/../../artifacts/moon_report.txt');
+        $content = file_get_contents(__DIR__ . $path);
```
Comment (warlof) [inline tests/ReportParser/Parsers/MoonReportTest.php:274]:
```suggestion
        $content = file_get_contents(__DIR__ . $artifact_path);
```

File: tests/ReportParser/Parsers/MoonReportTest.php [hunk 4]
```diff
@@ -202,19 +253,25 @@ public function testGetHeader()
...
         $report->parse($content);
 
         $this->assertTrue($report->hasHeader());
     }
 
-    public function testGetGroups()
+    /**
+     * @dataProvider correctData
+     */
+    public function testGetGroups(string $path)
```
Comment (warlof) [inline tests/ReportParser/Parsers/MoonReportTest.php:272]:
```suggestion
    public function testGetGroups(string $artifact_path)
```

File: tests/ReportParser/Parsers/MoonReportTest.php [hunk 5]
```diff
@@ -202,19 +253,25 @@ public function testGetHeader()
...
     }
 
-    public function testHasHeader()
+    /**
+     * @dataProvider correctData
+     */
+    public function testHasHeader(string $path)
     {
-        $content = file_get_contents(__DIR__ . '/../../artifacts/moon_report.txt');
+        $content = file_get_contents(__DIR__ . $path);
```
Comment (warlof) [inline tests/ReportParser/Parsers/MoonReportTest.php:261]:
```suggestion
        $content = file_get_contents(__DIR__ . $artifact_path);
```

File: tests/ReportParser/Parsers/MoonReportTest.php [hunk 6]
```diff
@@ -202,19 +253,25 @@ public function testGetHeader()
         ], $report->getHeader()->fields());
     }
 
-    public function testHasHeader()
+    /**
+     * @dataProvider correctData
+     */
+    public function testHasHeader(string $path)
```
Comment (warlof) [inline tests/ReportParser/Parsers/MoonReportTest.php:259]:
```suggestion
    public function testHasHeader(string $artifact_path)
```

File: tests/ReportParser/Parsers/MoonReportTest.php [hunk 7]
```diff
@@ -174,19 +219,25 @@ public function testParse()
...
     }
 
-    public function testGetHeader()
+    /**
+     * @dataProvider correctData
+     */
+    public function testGetHeader(string $path)
     {
-        $content = file_get_contents(__DIR__ . '/../../artifacts/moon_report.txt');
+        $content = file_get_contents(__DIR__ . $path);
```
Comment (warlof) [inline tests/ReportParser/Parsers/MoonReportTest.php:240]:
```suggestion
        $content = file_get_contents(__DIR__ . $artifact_path);
```

File: tests/ReportParser/Parsers/MoonReportTest.php [hunk 8]
```diff
@@ -57,9 +93,12 @@ public function testMissingReportHeaderException()
         $report->validate();
     }
 
-    public function testMissingReportGroupException()
+    /**
+     * @dataProvider malformedGroups
+     */
+    public function testMissingReportGroupException(string $path)
```
Comment (warlof) [inline tests/ReportParser/Parsers/MoonReportTest.php:99]:
```suggestion
    public function testMissingReportGroupException(string $artifact_path)
```

File: tests/ReportParser/Parsers/MoonReportTest.php [hunk 9]
```diff
@@ -46,9 +79,12 @@ public function testEmptyReportException()
         $report->validate();
     }
 
-    public function testMissingReportHeaderException()
+    /**
+     * @dataProvider malformedHeader
+     */
+    public function testMissingReportHeaderException(string $path)
```
Comment (warlof) [inline tests/ReportParser/Parsers/MoonReportTest.php:85]:
```suggestion
    public function testMissingReportHeaderException(string $artifact_path)
```

File: tests/ReportParser/Parsers/MoonReportTest.php [hunk 10]
```diff
@@ -202,19 +253,25 @@ public function testGetHeader()
         ], $report->getHeader()->fields());
     }
 
-    public function testHasHeader()
+    /**
+     * @dataProvider correctData
```
Comment (warlof) [inline tests/ReportParser/Parsers/MoonReportTest.php:257]:
```suggestion
     * @dataProvider correctFormatProvider
```

File: tests/ReportParser/Parsers/MoonReportTest.php [hunk 11]
```diff
@@ -202,19 +253,25 @@ public function testGetHeader()
...
 
         $report = new MoonReport();
         $report->parse($content);
 
         $this->assertTrue($report->hasHeader());
     }
 
-    public function testGetGroups()
+    /**
+     * @dataProvider correctData
```
Comment (warlof) [inline tests/ReportParser/Parsers/MoonReportTest.php:270]:
```suggestion
     * @dataProvider correctFormatProvider
```

File: tests/ReportParser/Parsers/MoonReportTest.php [hunk 12]
```diff
@@ -223,9 +280,12 @@ public function testGetGroups()
         $this->assertContainsOnlyInstancesOf(Group::class, $report->getGroups());
     }
 
-    public function testHasElements()
+    /**
+     * @dataProvider correctData
```
Comment (warlof) [inline tests/ReportParser/Parsers/MoonReportTest.php:284]:
```suggestion
     * @dataProvider correctFormatProvider
```

File: tests/ReportParser/Parsers/MoonReportTest.php [hunk 13]
```diff
@@ -13,9 +13,42 @@
...
     {
-        $content = file_get_contents(__DIR__ . '/../../artifacts/moon_report.txt');
+        return [
+            'from EVE'     => ['/../../artifacts/moon_report.txt'],
+            'from Excel'   => ['/../../artifacts/moon_report_excel.txt'],
+            'mixed inputs' => ['/../../artifacts/moon_report_mixed.txt']
+        ];
+    }
+
+    public function malformedElements(): array
```
Comment (warlof) [inline tests/ReportParser/Parsers/MoonReportTest.php:25]:
```suggestion
    public function malformedElementsProvider(): array
```

File: tests/ReportParser/Parsers/MoonReportTest.php [hunk 14]
```diff
@@ -46,9 +79,12 @@ public function testEmptyReportException()
         $report->validate();
     }
 
-    public function testMissingReportHeaderException()
+    /**
+     * @dataProvider malformedHeader
```
Comment (warlof) [inline tests/ReportParser/Parsers/MoonReportTest.php:83]:
```suggestion
     * @dataProvider malformedElementsProvider
```

File: tests/ReportParser/Parsers/MoonReportTest.php [hunk 15]
```diff
@@ -68,9 +107,12 @@ public function testMissingReportGroupException()
         $report->validate();
     }
 
-    public function testInvalidReportGroupException()
+    /**
+     * @dataProvider malformedElements
```
Comment (warlof) [inline tests/ReportParser/Parsers/MoonReportTest.php:111]:
```suggestion
     * @dataProvider malformedElementsProvider
```

File: tests/ReportParser/Parsers/MoonReportTest.php [hunk 16]
```diff
@@ -13,9 +13,42 @@
...
+    }
+
+    public function malformedElements(): array
+    {
+        return [
+            'no elements' => ['/../../artifacts/moon_report_without_elements.txt']
+        ];
+    }
+
+    public function malformedGroups(): array
```
Comment (warlof) [inline tests/ReportParser/Parsers/MoonReportTest.php:32]:
```suggestion
    public function malformedGroupsProvider(): array
```

File: tests/ReportParser/Parsers/MoonReportTest.php [hunk 17]
```diff
@@ -57,9 +93,12 @@ public function testMissingReportHeaderException()
         $report->validate();
     }
 
-    public function testMissingReportGroupException()
+    /**
+     * @dataProvider malformedGroups
```
Comment (warlof) [inline tests/ReportParser/Parsers/MoonReportTest.php:97]:
```suggestion
     * @dataProvider malformedGroupsProvider
```

File: tests/ReportParser/Parsers/MoonReportTest.php [hunk 18]
```diff
@@ -13,9 +13,42 @@
...
+    }
+
+    public function malformedGroups(): array
+    {
+        return [
+            'no groups' => ['/../../artifacts/moon_report_without_groups.txt']
+        ];
+    }
+
+    public function malformedHeader(): array
```
Comment (warlof) [inline tests/ReportParser/Parsers/MoonReportTest.php:39]:
```suggestion
    public function malformedHeaderProvider(): array
```

File: tests/ReportParser/Parsers/MoonReportTest.php (hunk 14 above)
Comment (warlof) [inline tests/ReportParser/Parsers/MoonReportTest.php:83]:
```suggestion
     * @dataProvider malformedHeaderProvider
```

**Review by warlof — CHANGES_REQUESTED** [review]
just a bit of wording and it will be ready for merge.
thank you very much :)

**Comment by peerau** [comment]
That should be a bit better for it

**Pull request merged**